microprofile.jdt/org.eclipse.lsp4mp.jdt.test/src/main/java/org/eclipse/lsp4mp/jdt/core/faulttolerance/java/MicroProfileFaultToleranceJavaDiagnosticsTest.java

```diff
@@ -88,6 +88,38 @@ public void asynchronousNonFutureOrCompletionStage() throws Exception {
 		assertJavaDiagnostics(diagnosticsParams, utils, d1, d2, d3);
 	}
 	
+	@Test
+	public void asynchronousClassNonFutureOrCompletionStage() throws Exception {
+		IJavaProject javaProject = loadMavenProject(MicroProfileMavenProjectName.microprofile_fault_tolerance);
+		IJDTUtils utils = JDT_UTILS;
+
+		MicroProfileJavaDiagnosticsParams diagnosticsParams = new MicroProfileJavaDiagnosticsParams();
+		IFile javaFile = javaProject.getProject()
+				.getFile(new Path("src/main/java/org/acme/AsynchronousFaultToleranceClassResource.java"));
+		diagnosticsParams.setUris(Arrays.asList(javaFile.getLocation().toFile().toURI().toString()));
+		diagnosticsParams.setDocumentFormat(DocumentFormat.Markdown);
+
+		Diagnostic d1 = d(32, 11, 17, 
+				"The annotated method does not return an object of type Future or CompletionStage", 
+				DiagnosticSeverity.Error,
+				MicroProfileFaultToleranceConstants.DIAGNOSTIC_SOURCE,
+				MicroProfileFaultToleranceErrorCode.FAULT_TOLERANCE_DEFINITION_EXCEPTION);
+		

```

Comment:
There's some extraneous whitespace in this file. You can usually see these if you run `git show` on your commit, in the commandline.

There's also an option in Eclipse to show whitespace characters and remove extraneous ones.